open-feature/dart-server-sdk
# feat(core): clean up providers during shutdown and enable transaction context propagation

## This PR
- Adds graceful provider cleanup during application shutdown.
- Implements transaction context propagation for evaluation contexts such as HTTP requests or threads.



### Notes
- Provider shutdown ensures that resources are cleaned up properly when the application stops.
- Transaction context propagation improves context management for multi-threaded or HTTP request-based workloads.

### Follow-up Tasks
- Add more testing coverage for edge cases in context propagation.
- Update documentation to reflect these changes.

### How to Test
1. Start the application and initialize multiple providers.
2. Trigger application shutdown (e.g., stop the server).
3. Observe if providers are cleaned up without errors.
4. Verify context propagation works in scenarios like HTTP requests or multi-threaded tasks.
